include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1101 @@
+     * after exceptions is thrown.
+     */
+    void resize(size_type const count)
+    {
+        assert(count < max_size());
+        data.resize(count);
+    }
+
+    /*!\copybrief resize()
+     * \param value Append copies of value when resizing.
+     * \copydetails resize()
+     */
+    void resize(size_type const count, value_type const value)
+    {
+        assert(count < max_size());
+        data.resize(count, to_rank(value));
+    }
+
+    /*!\brief Swap contents with another instance.
+     * \param rhs The other instance to swap with.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr void swap(bitcompressed_vector & rhs) noexcept
+    {
+        std::swap(data, rhs.data);
+    }
+
+    //!\copydoc swap()
+    constexpr void swap(bitcompressed_vector && rhs) noexcept
+    {
+        std::swap(data, rhs.data);
+    }
+    //!\}
+
+    /*!\brief Swap contents with another instance.
+     * \param lhs The first instance.
+     * \param rhs The other instance to swap with.
+     *
+     * ### Complexity
+     *
+     * Constant.
+     *
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    friend constexpr void swap(bitcompressed_vector & lhs, bitcompressed_vector & rhs) noexcept
+    {
+        std::swap(lhs, rhs);
+    }
+
+    //!\overload
+    friend constexpr void swap(bitcompressed_vector && lhs, bitcompressed_vector && rhs) noexcept
+    {
+        std::swap(lhs, rhs);
+    }
+    //!\}
+
+    //!\name Comparison operators
+    //!\{
+    constexpr bool operator==(bitcompressed_vector const & rhs) const noexcept
+    {
+        return data == rhs.data;
+    }
+
+    constexpr bool operator!=(bitcompressed_vector const & rhs) const noexcept
+    {
+        return data != rhs.data;
+    }
+
+    constexpr bool operator<(bitcompressed_vector const & rhs) const noexcept
+    {
+        return data < rhs.data;
+    }
+
+    constexpr bool operator>(bitcompressed_vector const & rhs) const noexcept
+    {
+        return data > rhs.data;
+    }
+
+    constexpr bool operator<=(bitcompressed_vector const & rhs) const noexcept
+    {
+        return data <= rhs.data;
+    }
+
+    constexpr bool operator>=(bitcompressed_vector const & rhs) const noexcept
+    {
+        return data >= rhs.data;
+    }
+    //!\}
+
+    /*!\cond DEV
+     * \brief Serialisation support function.
+     * \tparam archive_t Type of `archive`; must satisfy seqan3::cereal_archive_concept.
+     * \param archive The archive being serialised from/to.
+     *
+     * \attention These functions are never called directly, see \ref serialisation for more details.
+     */
+    template <cereal_archive_concept archive_t>
+    void CEREAL_SERIALIZE_FUNCTION_NAME(archive_t & archive)
+    {
+        archive(data); //TODO: data not yet serialisable
```

Comment:
please add a ticket for this and add it to the seqan development project.

Reply:
done